### PR TITLE
fix(button base): dont set href if disabled is true

### PR DESCRIPTION
**Pull request by saghdaey**
### Overview

Allow links to be disabled by not setting href attribute if disabled prop is true

Context: 
I'd like to disable a link to the workspaces/new page if the user is not verified
I passed in disabled to the gamut button but bc the component did have an href, the button was still clickable despite appearing disabled  
https://github.com/codecademy-engineering/Codecademy/pull/28066#issuecomment-962027582

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: EGG-1981
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/ButtonBase/ButtonBase.tsx [hunk 1]
```diff
@@ -85,7 +86,6 @@ export const ButtonBase = forwardRef<
         ref={ref as MutableRefObject<HTMLAnchorElement>}
         as="a"
         href={href}
-        aria-disabled={!!disabled}
```
Comment (saghdaey) [inline packages/gamut/src/ButtonBase/ButtonBase.tsx:88]: removed this bc I dont think we'll ever end up at this line given change to line 67?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.8-alpha.60ac54.0<br/>@codecademy/gamut-labs@28.0.3-alpha.60ac54.0<br/>@codecademy/gamut@40.0.1-alpha.60ac54.0<br/>@codecademy/monaco@0.5.6-alpha.60ac54.0<br/>@codecademy/styleguide@50.1.5-alpha.60ac54.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618570eaf24ded0c47ece3d1--gamut-preview.netlify.app>https://618570eaf24ded0c47ece3d1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618570eaf24ded0c47ece3d1>Deploy Logs</a></p>

File: packages/gamut/src/ButtonBase/ButtonBase.tsx [hunk 2]
```diff
@@ -64,7 +64,7 @@ export const ButtonBase = forwardRef<
     { href, disabled, children, role = 'button', type = 'button', ...rest },
     ref
   ) => {
-    if (href == null) {
+    if (href == null || disabled) {
```
Comment (saghdaey) [inline packages/gamut/src/ButtonBase/ButtonBase.tsx:67]: basically i'm proposing we return a disabled button if an href is provided but disabled is true, that way we prevent the button from being clickable.. lemme know if that's bad?

File: packages/gamut/src/ButtonBase/ButtonBase.tsx [hunk 3]
```diff
@@ -73,6 +73,7 @@ export const ButtonBase = forwardRef<
           type={type}
           role={role}
           disabled={!!disabled}
+          aria-disabled={!!disabled}
```
Comment (saghdaey) [inline packages/gamut/src/ButtonBase/ButtonBase.tsx:76]: do we need both?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.8-alpha.311079.0<br/>@codecademy/gamut-labs@28.0.3-alpha.311079.0<br/>@codecademy/gamut@40.0.1-alpha.311079.0<br/>@codecademy/monaco@0.5.6-alpha.311079.0<br/>@codecademy/styleguide@50.1.5-alpha.311079.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61857cc28720d81cb6116fda--gamut-preview.netlify.app>https://61857cc28720d81cb6116fda--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61857cc28720d81cb6116fda>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.9-alpha.7dd2d4.0<br/>@codecademy/gamut-labs@28.0.4-alpha.7dd2d4.0<br/>@codecademy/gamut@40.1.1-alpha.7dd2d4.0<br/>@codecademy/monaco@0.5.7-alpha.7dd2d4.0<br/>@codecademy/styleguide@50.1.6-alpha.7dd2d4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61858950772100252ae1a65e--gamut-preview.netlify.app>https://61858950772100252ae1a65e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61858950772100252ae1a65e>Deploy Logs</a></p>

**Review by hmafzal — APPROVED** [review]
LGTM

File: packages/gamut/src/ButtonBase/ButtonBase.tsx (hunk 2 above)
Comment (hmafzal) [reply packages/gamut/src/ButtonBase/ButtonBase.tsx:67]: Seems reasonable to me. 

https://css-tricks.com/how-to-disable-links/

This article seems to agree around removing the href. Switching to a button gives us the advantage of having the correct functionality we would want for disabled behavior without having to add additional styling/functionality to the`<a>` element

Comment (saghdaey) [reply packages/gamut/src/ButtonBase/ButtonBase.tsx:67]: this is also what i looked at!! :) i love css-tricks

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.10-alpha.46aa39.0<br/>@codecademy/gamut-labs@28.0.5-alpha.46aa39.0<br/>@codecademy/gamut@40.1.2-alpha.46aa39.0<br/>@codecademy/monaco@0.5.8-alpha.46aa39.0<br/>@codecademy/styleguide@50.1.7-alpha.46aa39.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618591d7e9b9f1009fe52e91--gamut-preview.netlify.app>https://618591d7e9b9f1009fe52e91--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618591d7e9b9f1009fe52e91>Deploy Logs</a></p>

File: packages/gamut/src/ButtonBase/__tests__/ButtonOutline-test.tsx [hunk 4]
```diff
@@ -31,7 +31,7 @@ describe('ButtonBase', () => {
   });
 
   describe('when the button is disabled', () => {
-    it('renders the aria-disabled attribute on anchor tags', () => {
+    it('does not render the href attribute if disabled and renders a button', () => {
```
Comment (dreamwasp) [inline packages/gamut/src/ButtonBase/__tests__/ButtonOutline-test.tsx:34]: it looks like these test are still failing -- will approve once that's cleared up!

File: packages/gamut/src/ButtonBase/ButtonBase.tsx (hunk 3 above)
Comment (dreamwasp) [reply packages/gamut/src/ButtonBase/ButtonBase.tsx:76]: i don't [think so](https://stackoverflow.com/questions/38059140/when-to-use-the-disabled-attribute-vs-the-aria-disabled-attribute-for-html-eleme)?

File: packages/gamut/src/ButtonBase/__tests__/ButtonOutline-test.tsx (hunk 4 above)
Comment (saghdaey) [reply packages/gamut/src/ButtonBase/__tests__/ButtonOutline-test.tsx:34]: great! thanks, will fix today :)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.10-alpha.313366.0<br/>@codecademy/gamut-labs@28.0.5-alpha.313366.0<br/>@codecademy/gamut@40.1.2-alpha.313366.0<br/>@codecademy/monaco@0.5.8-alpha.313366.0<br/>@codecademy/styleguide@50.1.7-alpha.313366.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61894e1f7965fe008581ac6d--gamut-preview.netlify.app>https://61894e1f7965fe008581ac6d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61894e1f7965fe008581ac6d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.10-alpha.cb8fbd.0<br/>@codecademy/gamut-labs@28.0.5-alpha.cb8fbd.0<br/>@codecademy/gamut@40.1.2-alpha.cb8fbd.0<br/>@codecademy/monaco@0.5.8-alpha.cb8fbd.0<br/>@codecademy/styleguide@50.1.7-alpha.cb8fbd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61895f4141f13912713f603a--gamut-preview.netlify.app>https://61895f4141f13912713f603a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61895f4141f13912713f603a>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]

**Pull request merged**